app/src/main/java/com/android/sample/ui/settings/SettingsScreen.kt

```diff
@@ -0,0 +1,108 @@
+package com.android.sample.ui.settings
+
+import android.widget.Toast
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.ArrowBack
+import androidx.compose.material.icons.outlined.Delete
+import androidx.compose.material.icons.outlined.ExitToApp
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.sample.ui.navigation.NavigationActions
+import com.google.firebase.auth.FirebaseAuth
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SettingsScreen(navigationActions: NavigationActions) {
+  val context = LocalContext.current
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = { Text("Settings", fontSize = 20.sp) },
+            navigationIcon = {
+              IconButton(onClick = { navigationActions.goBack() }) {
+                Icon(imageVector = Icons.Outlined.ArrowBack, contentDescription = "Back")
+              }
+            },
+            modifier = Modifier.testTag("settingsScreen") // Add testTag for the screen itself
+            )
+      },
+      content = { paddingValues ->
+        Column(
+            modifier = Modifier.fillMaxSize().padding(paddingValues).padding(horizontal = 24.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              // User data button
+              Button(
+                  onClick = { /* TODO: Handle User Data */},
+                  modifier =
+                      Modifier.fillMaxWidth()
+                          .height(60.dp)
+                          .background(Color.LightGray, RoundedCornerShape(10.dp))
+                          .testTag("userDataButton"), // Test tag for User Data button
+                  colors = ButtonDefaults.buttonColors(containerColor = Color.LightGray)) {
+                    Text("User data", color = Color.Black)
+                  }
+
+              // Preferences button
+              Button(
+                  onClick = { /* TODO: Handle Preferences */},
+                  modifier =
+                      Modifier.fillMaxWidth()
+                          .height(60.dp)
+                          .background(Color.LightGray, RoundedCornerShape(10.dp))
+                          .testTag("preferencesButton"), // Test tag for Preferences button
+                  colors = ButtonDefaults.buttonColors(containerColor = Color.LightGray)) {
+                    Text("Preferences", color = Color.Black)
+                  }
+
```

Comment:
The buttons are good example for when extracting a common composable is useful. It makes it easy to have the same look across the whole app while being able to quickly change it if needed.